designs/static-capacity.md

```diff
@@ -0,0 +1,202 @@
+
+```yaml
+spec:
+  # Makes NodePool static when specified
+  replicas: 5  # Number of nodes to maintain
+limits:
+  # Prevents Karpenter from creating new replicas once the limit is exceeded
+  nodes: 10 # Constrain the size of static Nodepool 
+```
+The existence of replicas field will determine if NodePool is Static
+
+Key aspects:
+1. Static NodePools maintain fixed node count regardless of pod demand
+2. Will not be considered as consolidation candidate, so consolidation settings are not allowed other than defaults (at least in v1alpha1)
+3. Optional limits.nodes can be set to constrain the size of static Nodepool
+4. Nodes are spread based on the topology requirements
+5. Inherit existing Karpenter features
+6. Use existing NodeClaim infrastructure with owner references
+  
+
+### Modeling & Validation
+
+When `replicas` is specified, the `NodePool` enters a static provisioning mode where certain disruption-related fields become irrelevant or misleading. Specifically:
+
+- `limits.cpu` and `limits.memory` (used to influence consolidation and node selection) are not meaningful for static capacity.
+- `consolidationPolicy` and `consolidateAfter` (which control Karpenter’s consolidation logic) must not be used when the node count is fixed via `replicas`.
+
+If the above values are specified it will simply be ignored. 
+
+Once a NodePool has replicas set, it cannot be removed unless the NodePool itself is deleted. This means a NodePool cannot switch between static and dynamic modes:
+    - If a NodePool is static (replicas set), replicas cannot be unset or modified to nil.
+    - If a NodePool is dynamic (replicas not set), replicas cannot be added.
+    - This validation prevents unintentional large-scale terminations or sudden behavior changes in consolidation logic.
+
+
+#### Validation rules
+
+There will not be a validation against consolidation settings (consolidationPolicy and consolidateAfter) and limits.cpu and limits.memory, if specified it will simply be ignored
+
+```yaml
+If has(self.replicas) != has(oldSelf.replicas)
+  => Validation Error
+```
+### Disruption
+
+Karpenter already has Disruption for nodes we can use the same mechanism. 
+Since Static Nodepool and NodeClaim is a variant of existing Nodepool/NodeClaim it will inherit Karpenter dynamic provider integration. Drift Detection can also be inherited to trigger replacement of drifted nodes. During this Karpenter will respect the Disruption Budget. If limits.nodes are specified then Karpenter would wait until Nodes are terminated and count is well within node limits before creating replacements.
+
+Replacement will be one-for-one i.e create replacements before terminating the old one to honor replicas if Nodepool node count do not exceed the node limits. 
+When a user scales down a static NodePool, Karpenter will forcefully terminate the corresponding NodeClaims/instances.
```

Comment:
I had some other comment regarding this, but I disagree with this being the behavior. Customer's have levers to ensure that nodes are terminated in a set period of time, they can continue to leverage them with static NodePools. I don't see any reason to create this API divergence.